src/plugin/customParseFormat/index.js

```diff
@@ -221,11 +231,18 @@ export default (o, C, d) => {
       if (!isStrictWithoutLocale && pl) {
         locale = d.Ls[pl]
       }
-      this.$d = parseFormattedInput(date, format, utc)
+      const { parsedDate, parsedOffsetMilliseconds } = parseFormattedInput(date, format, utc)
+      this.$d = parsedDate
       this.init()
       if (pl && pl !== true) this.$L = this.locale(pl).$L
-      if (isStrict && date !== this.format(format)) {
-        this.$d = new Date('')
+      const currentOffset = this.utcOffset()
+      if (isStrict) {
+        if (
+          (parsedOffsetMilliseconds !== undefined && date.replace(/[+-]\d\d:?\d\d$/, '') !== this.subtract(currentOffset, 'minute').subtract(parsedOffsetMilliseconds, 'millisecond').format(format.replace(/(ZZ|ZZZ)$/, '')))
```

Comment:
Why is `ZZZ` (triple zed) here? I don't see in the docs: https://day.js.org/docs/en/parse/string-format